embedded-graphics/src/pixelcolor/mod.rs

```diff
@@ -1,25 +1,101 @@
+//!
+//! /// Color with 3 states.
+//! #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+//! pub enum EpdColor {
+//!     White,
+//!     Black,
+//!     Red,
+//! }
+//!
+//! /// The `Raw` can be is set to `()` because `EpdColor` doesn't need to be
+//! /// converted to raw data for the display and isn't stored in images.
+//! impl PixelColor for EpdColor {
+//!     type Raw = ();
+//! }
+//!
+//! /// Mock EPD display.
+//! pub struct EpdDisplay {}
+//!
+//! impl Drawing<EpdColor> for EpdDisplay {
+//!     fn draw<T>(&mut self, item: T)
+//!     where
+//!         T: IntoIterator<Item = Pixel<EpdColor>>
+//!     {
+//!         for Pixel(point, color) in item {
+//!             match color {
+//!                 EpdColor::White => {} // draw white pixel at `point`
+//!                 EpdColor::Black => {} // draw black pixel at `point`
+//!                 EpdColor::Red => {} // draw red pixel at `point`
+//!             }
+//!         }
+//!     }
+//! }
+//!
+//! fn main() {
+//!     let mut display = EpdDisplay {};
+//!
+//!     display.draw(
+//!         egrectangle!((0, 0), (100, 100), fill = Some(EpdColor::White))
+//!     );
+//!
+//!     display.draw(
+//!         egrectangle!((100, 0), (200, 100), fill = Some(EpdColor::Black))
+//!     );
+//!
+//!     display.draw(
+//!         egrectangle!((200, 0), (300, 100), fill = Some(EpdColor::Red))
+//!     );
+//! }
+//! ```
+//!
+//! The implementation of the [`Drawing`] trait uses a `match` statement to
+//! draw the specified color and doesn't depend on the raw data conversions,
+//! see the [`raw` module] documentation for an example that uses this feature.
+//!
+//! [`Drawing`]: ../trait.Drawing.html
+//! [`PixelColor`]: trait.PixelColor.html
+//! [`Raw`]: trait.PixelColor.html#associatedtype.Raw
+//! [`raw` module]: raw/index.html
 
 mod binary_color;
 mod conversion;
 mod gray_color;
+pub mod raw;
 mod rgb_color;
 
 pub use binary_color::*;
 pub use gray_color::*;
 pub use rgb_color::*;
 
-use core::fmt;
-
-/// Pixel color trait
-pub trait PixelColor: Clone + Copy + PartialEq + fmt::Debug {}
-
-/// Convert raw data to color structs.
+/// Pixel color trait.
+///
+/// This trait is used to represent types that can be used in drawing methods.
+///
+/// See the [module-level documentation] for more details.
 ///
-/// This trait is used to generically convert raw color data with different
-/// bitdepth to specific color types. If the number of bits required to build
-/// the color is less than 32 the upper bits are ignored.
-pub trait FromRawData {
-    /// Convert raw data to color.
-    fn from_raw_data(data: u32) -> Self;
+/// [module-level documentation]: index.html
+pub trait PixelColor: Copy + PartialEq {
+    /// Raw data type.
+    ///
+    /// Specifies the raw storage type that can be used to represent this color.
+    ///
+    /// See the [`raw` module documentation] for more details.
+    ///
+    /// [`raw` module documentation]: raw/index.html
+    type Raw: raw::RawData;
```

Comment:
I'm not sure if `Storage` should be used here, because `Raw` isn't stored inside the `PixelColor`. But let's decide on a name for the `RawUx` types first.